dev/planning-example.py

```diff
@@ -0,0 +1,265 @@
+
+# initializing from astropy entities directly
+from astropy.coordinates import SkyCoord
+
+t1 = SiderealTarget(name='Polaris',
+               coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs'))
+
+# Leaves scope for NonSiderealTarget, etc.
+
+# Convenience methods for looking up/constructing targets by name via
+# astroquery?
+
+
+# ================================
+# Condition objects, observability
+# ================================
+
+"""
+Q: Can I observe this target on the night of May 1, 2015 between 18:30
+and 05:30 HST for 30 minutes somewhere between 15 degrees and 89 degrees
+altitude?
+"""
+
+# first we define the conditions
+from astroplan.entity import Constraints
+
+# times can be passed in as strings (interpreted as for get_date()) or
+# as astropy Time or datetime objects
+cts = Constraints(time_start="2015-05-01 18:30", time_end="2015-05-02 05:30",
+                  alt_min=15.0*u.deg, alt_max=89.0*u.deg, duration=30*u.min)
+
+# define a target
+tgt = SiderealTarget(name='S5', ra='14:20:00.00', dec='48:00:00.00')
+
+# returns an object with information about the observability with these
+# constraints
+info = obs.observable(tgt, cts)
+
+'''
+attributes of return object:
+  observable: a boolean -- whether target is observable at desired constraints
+  rise_time: a datetime -- first time target is visible in at these conditions
+  set_time: a datetime -- time target is no longer visible at these conditions
+
+Use of a return object means that additional items can be added in the
+future without breaking the API (e.g. subclasses are free to attach
+additional items).  It can be also useful to attach an "explanation" for why
+something cannot be observed.
+'''
+
+# same question, but specify the lower elevation bound to be the
+# telescope minimum and I will specify an airmass of 1.2 for the target
+cond = Constraints(time_start="2015-05-01 18:30", time_end="2015-05-02 05:30",
+                   alt_min=15.0*u.deg, alt_max=89.0*u.deg, duration=30*u.min,
+                   airmass=1.2)
+info = obs.observable(tgt, cond)
+
+'''
+Suggest that Constraints implement a time range, an altitude range and an
+airmass limit at beginning.  Other things that could be added later:
+ - azimuth range
+ - airmass range
+
+Idea is that Constraints can be subclassed to add additional constraints.
+'''
+
+# ======================================================
+# Other useful calculations wrt an observer and a target
+#=======================================================
+
+# calculate the distance in alt and az degrees between two targets at
+# the given time (e.g. to calculate slew time)
+sf = SiderealTarget(name='Sf', ra='09:40:00.00', dec='43:00:00.00')
+sm = SiderealTarget(name='Sm', ra='10:30:00.00', dec='36:00:00.00')
+t = obs.get_date("2015-05-01 23:00")
+
+# returns a vector or other astropy quantity specifying a delta in alt, az
+obs.distance(sf, sm, t)
```

Comment:
astropy also uses ``separation`` to mean "angular distance between two points``, so that's a bit problematic an that it's got an overloaded meaning.  Anything as short as "distance" or "separation" has these problems, actually.

And that sort of ambiguity is why I'm actually not convinced this should exist at all...  This goes back to a discussion we had with astropy.coordinates: if you want the delta-RA and delta-Dec between two coordinates, you could imagine something like ``dra, ddec = c1.deltaradec(c2)``.  But it turns out that some people will take that to mean ``c1-c2`` and others will think it's ``c2-c1``.  So instead we decided that expecting users to do ``dra, ddec = c1.ra - c2.ra, c1.dec - c2.dec`` wasn't too onerous.  Similarly, this could be implemented as something like ``dalt = obs.altaz(sf, t).alt - obs.altaz(sm, t).alt``, with less "magic" under the hood and repetition.

But if y'all don't like that, another option would be to try a more specific name like ``obs.daltaz(sf, sm, t)``